vdev/src/util.rs

```diff
@@ -37,15 +37,28 @@ pub fn git_head() -> Result<Output> {
         .context("Could not execute `git`")
 }
 
-/// Calculate the release channel from `git describe`
-pub fn release_channel() -> Result<&'static str> {
-    git_head().map(|output| {
-        if output.status.success() {
-            "latest"
-        } else {
-            "nightly"
+// If in CI, it is necessary to adjust the permissions on the git repository
+// as a precursor to running some git commands.
+pub fn mark_safe_git_repo() {
+    if let Ok(is_ci) = std::env::var("CI") {
+        if is_ci == "true" {
+            std::process::Command::new("git")
```

Comment:
My thought process was that the CI build workflows are tightly coupled to vdev already (the Makefile targets that the CI workflow invokes directly call `vdev package <...>`), and that this was a more centralized location for the logic, than a new step in each of the specific packaging jobs, because we also have to do it before running the vdev "version" subcommand.
The command is gated on CI env var so it won't be executed in local vdev usage.

Reply:
chatted with Jesse about this out of band. I will try pulling this out of vdev into a pre step in CI, and if issues arise which led to the placement in vdev I can surface that.

We may decide to make vdev more CI aware but that doesn't have to be in scope for this endeavor.